streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -162,7 +172,13 @@ public KStreamImpl(final InternalStreamsBuilder builder,
         Objects.requireNonNull(mapper, "mapper can't be null");
         String name = builder.newProcessorName(MAP_NAME);
 
-        builder.internalTopologyBuilder.addProcessor(name, new KStreamMap<>(mapper), this.name);
+        ProcessorParameters processorParameters = new ProcessorParameters<>(new KStreamMap<>(mapper), name);
+
+        StatelessProcessorNode<K1, V1> mapProcessorNode = new StatelessProcessorNode<>(name,
+                                                                                       processorParameters,
+                                                                                       true);
+        mapProcessorNode.setTriggersRepartitioning(true);
```

Comment:
Hmm.. why we decide that we will definitely trigger a rebalance at this node without finishing building the stream graph?

Reply:
this is just metadata used for a hint that this node will trigger a repartition and this information will be used to determine if possible optimizations are available.